lib/logstash/codecs/json_lines.rb

```diff
@@ -55,6 +57,9 @@ class LogStash::Codecs::JSONLines < LogStash::Codecs::Base
   public
 
   def register
+    if decode_size_limit_bytes == DEFAULT_DECODE_SIZE_LIMIT_BYTES
```

Comment:
This will show a warning if a user happens to set the value to 512mb, it's not what we want.
We want to show the deprecation notice if the user leaves this as default and doesn't set the value, something like:

```suggestion
    if original_params['decode_size_limit_bytes'].nil?
```

(haven't confirmed this works)

Reply:
`original_params` contains the full configuration.
This is the content when I don't set anything:
```
{"id"=>"17986701-2b7f-49b7-ac56-eff63b1ea6b3", "enable_metric"=>true, "charset"=>"UTF-8", "delimiter"=>"\n", "decode_size_limit_bytes"=>536870912}
```

while this is the one when when I configure the plugin with:
```
codec => json_lines {
  decode_size_limit_bytes => 32768
}
```

So apparently there is not way to distinguish if a user set `decode_size_limit_bytes` to default value or is effetively the setting untouched.

Reply:
At a quick glance at the code this doesn't seem to be true, I can see the original_params only having the ID. I do see however that there are two register calls, and when the second one happens the original_params is no longer "original". this seems to be a bug, maybe in how we clone codec instances.
Please confirm the behaviour. if original_params is indeed equal to "params" then we have a larger bug we need to fix, as that was the whole purpose of "original_params"

Reply:
The problem related to receiving `params` instead of `original_params` in the second call was moved to https://github.com/elastic/logstash/issues/16425